chain/ethevents/handlers.go

```diff
@@ -141,10 +142,66 @@ func HandleVochainOracle(ctx context.Context, event *ethtypes.Log, e *EthereumEv
 
 		res, err := e.VochainApp.SendTX(tx)
 		if err != nil || res == nil {
-			log.Warnf("cannot broadcast tx: (%s)", err)
 			return fmt.Errorf("cannot broadcast tx: %w, res: %+v", err, res)
 		}
-		log.Infof("oracle transaction sent, hash: %s", res.Hash)
+		log.Infof("oracle transaction sent, hash: %x", res.Hash)
+
+	case ethereumEventList["processesCensusUpdated"]:
+		tctx, cancel := context.WithTimeout(ctx, time.Minute)
+		defer cancel()
+		setProcessTx, err := processCensusUpdatedMeta(tctx, &e.ContractsABI[0], event.Data, e.VotingHandle)
+		if err != nil {
+			return fmt.Errorf("cannot obtain process data for creating the transaction: %w", err)
+
+		}
+		log.Infof("found process %x census update on ethereum", setProcessTx.ProcessId)
+		p, err := e.VochainApp.State.Process(setProcessTx.ProcessId, true)
+		if err != nil {
+			return fmt.Errorf("cannot fetch the process from the Vochain: %w", err)
+		}
+
+		// process censusURI
+		if bytes.Equal(p.CensusRoot, setProcessTx.CensusRoot) {
+			return fmt.Errorf("censusRoot cannot be the same")
+		}
+		// check origin
+		if p.CensusURI == setProcessTx.CensusURI {
+			return fmt.Errorf("censusURI cannot be the same")
```

Comment:
I don't think this check should be done. The censusURI might be the same. I suppose you are thinking on a IPFS census but we would extend it to HTTP(s) or other URL typey.